worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,501 @@
+        if change:
+            # Orbsanity as an option is no-factor to these conditions. Matching the item name implies Orbsanity is ON,
+            # so we don't need to check the option. When Orbsanity is OFF, there won't even be any orb bundle items
+            # to collect.
+
+            # Orb items do not intrinsically unlock anything that contains more Reachable Orbs, so they do not need to
+            # set the cache to stale. They just change how many orbs you have to trade with.
+            if item.name == self.orb_bundle_item_name:
+                state.prog_items[self.player]["Tradeable Orbs"] += self.orb_bundle_size  # Give a bundle of Trade Orbs
+
+            # Scout Flies ALSO do not unlock anything that contains more Reachable Orbs, NOR do they give you more
+            # tradeable orbs. So let's just pass on them.
+            elif item.name in self.item_name_groups["Scout Flies"]:
+                pass
+
+            # Power Cells DO unlock new regions that contain more Reachable Orbs - the connector levels and new
+            # hub levels - BUT they only do that when you have a number of them equal to one of the threshold values.
+            elif (item.name == "Power Cell"
+                  and state.count("Power Cell", self.player) not in self.power_cell_thresholds):
+                pass
+
+            # However, every other item that changes the CollectionState should set the cache to stale, because they
+            # likely made it possible to reach more orb locations (level unlocks, region unlocks, etc.).
+            else:
+                state.prog_items[self.player]["Reachable Orbs Fresh"] = False
+        return change
+
+    def remove(self, state: CollectionState, item: Item) -> bool:
+        change = super().remove(state, item)
+        if change:
+
+            # Do the same thing we did in collect, except subtract trade orbs instead of add.
+            if item.name == self.orb_bundle_item_name:
+                state.prog_items[self.player]["Tradeable Orbs"] -= self.orb_bundle_size  # Take a bundle of Trade Orbs
+
+            # Ditto Scout Flies.
+            elif item.name in self.item_name_groups["Scout Flies"]:
+                pass
+
+            # Ditto Power Cells, but check count + 1, because we potentially crossed the threshold in the opposite
+            # direction. E.g. we've removed the 20th power cell, our count is now 19, so we should stale the cache.
+            elif (item.name == "Power Cell"
+                  and state.count("Power Cell", self.player) + 1 not in self.power_cell_thresholds):
+                pass
+
+            # Ditto everything else.
+            else:
+                state.prog_items[self.player]["Reachable Orbs Fresh"] = False
+        return change
+
+    def fill_slot_data(self) -> dict[str, Any]:
+        slot_dict = {
+            "slot_name": self.player_name,
+            "slot_seed": self.multiworld.seed_name,
```

Comment:
Both of these in the slot_dict are duplicates of data that already exists on the network protocol.

Reply:
For context: I use this data to populate some in-game save data information. Jak has 4 save slots, so I have a structure that saves the player name and seed to the slot you choose when you start a new game. This helps players figure out which slot to load when returning to an async game.

![image](https://github.com/user-attachments/assets/315c4915-6d1a-442a-bf5c-dc0fe22b27b9)

I'm guessing from the network doc that `seed_name` is on RoomInfo packets and `player_name` can be derived from info on Connected packets. Since I need both pieces of info before I send it to the game, which of these packets should I do that on, i.e. which comes last?

Reply:
Fixed.